### ISSUE TITLE
[STOREKIT2_MODE] getPurchaseHistory but the result is an empty array

### ISSUE THREAD

**Issue by 19424056** [dooboolab/react-native-iap#2109]
> Please use the [Discussion board](https://github.com/dooboolab/react-native-iap/discussions) if you want to get some help. Please use issues to report bugs.

**Description**
storekitMode: 'STOREKIT2_MODE'

Code i am using:
```
    const response = await getAvailablePurchases({
      alsoPublishToEventListener: false,
      automaticallyFinishRestoredTransactions: false,
      onlyIncludeActiveItems: false,
    });
    
   console.log(response) // []
```

getPurchaseHistory but current using function getAvailableItems, so when call getPurchaseHistory then return transactions available

https://developer.apple.com/documentation/storekit/transaction/3851204-currententitlements
![image](https://user-images.githubusercontent.com/60703498/201486605-1b4b5b3b-d81c-4e23-845e-183dbdf6d2b3.png)

node_modules/react-native-iap/src/iap.ts
![image](https://user-images.githubusercontent.com/60703498/201486641-a903ff14-f61a-4e5d-b7a4-302021782901.png)

node_modules/react-native-iap/ios/RNIapIosSk2.swift
![image](https://user-images.githubusercontent.com/60703498/201486706-66e75a59-c4f0-4ade-a627-d88959b3f29d.png)

I think the problem is here, i tried it and it works well:
https://developer.apple.com/documentation/storekit/transaction/3851203-all
![image](https://user-images.githubusercontent.com/60703498/201486736-1a48cb50-2e41-4884-9016-4fa64f4d284c.png)

node_modules/react-native-iap/ios/RNIapIosSk2.swift
add new function getHistoryItems
![image](https://user-images.githubusercontent.com/60703498/201486777-4e33262c-bb28-4a6d-870a-bb6901e779fa.png)

node_modules/react-native-iap/src/iap.ts
![image](https://user-images.githubusercontent.com/60703498/201486754-f06abd5e-7ebf-4634-93a8-ed63813bd438.png)

**Expected Behavior**
The string is printed to the console

**Screenshots**
N/A

**Environment:**

- react-native-iap: 12.4.4
- react-native: 0.70.5
- Platforms (iOS, Android, emulator, simulator, device): IOS(STOREKIT2_MODE)

**To Reproduce**

1. Paste code within the requestPurchase function
2. On device make getPurchaseHistory

---

[Optional] **Additional Context**

**Comment by 19424056** [comment]
@andresesfm Do you think that is the cause? Please help me

**Comment by andresesfm** [comment]
@19424056 Thank you for your feedback. I can see the need for all the transactions to be returned. Here is a proposed solution: https://github.com/dooboolab/react-native-iap/pull/2134

**Comment by andresesfm** [comment]
Released in `12.4.6`. Let me know/reopen if any issues
